fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,272 @@
+        if not all(ele >= 0 for ele in tup):
+            raise ValueError(_ERROR_BARS_NEGATIVE_VALUE_ERROR)
+
+
+def _check_valid_conf_interval(df_conf_intervals):
+    for tup in df_conf_intervals:
+        if not isinstance(tup, tuple) or len(tup) != 2:
+            raise ValueError(_CONF_INTERVALS_MUST_BE_TUPLE)
+        if tup[0] > tup[1]:
+            raise ValueError(_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR)
+
+
+def _plot_df(df, metrics, kind, subplots, df_all_errors=None, **kwargs):
+    if df_all_errors is not None:
+        yerr = np.array(
+            [np.array([[*row] for row in df_all_errors[metric]]).T for metric in metrics]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', yerr=yerr,
+                                   subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, yerr=yerr, subplots=subplots, **kwargs)
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                if kind == "scatter":
+                    color = ax.lines[0].get_color()
+                else:
+                    color = "black"
+
+                # extend legend with 95% CI text
+                handles, labels = ax.get_legend_handles_labels()
+                custom_line = [Line2D([0], [0], color=color, label="95% CI")]
+                handles.extend(custom_line)
+                ax.legend(handles=handles)
+        else:
+            if kind == "scatter":
+                color = axs.lines[0].get_color()
+            else:
+                color = "black"
+
+            # extend legend with 95% CI text
+            handles, labels = axs.get_legend_handles_labels()
+            custom_line = [Line2D([0], [0], color=color, label="95% CI")]
+            handles.extend(custom_line)
+            axs.legend(handles=handles)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(metric_frame: MetricFrame,
+                      kind: str = "scatter",
+                      metrics: Union[List[str], str] = None,
+                      error_bars: Union[List[str], str] = None,
+                      conf_intervals: Union[List[str], str] = None,
+                      subplots: bool = True,
+                      plot_error_labels: bool = False,
+                      text_precision_digits: int = 4,
+                      text_fontsize: int = 8,
+                      text_color: str = "black",
+                      text_ha: str = "center",
+                      **kwargs
+                      ) -> Union[Axes, List[Axes]]:
+    """Visualization for metrics with statistical error bounds.
+
+    Plots a given metric and its given error (as described by the `error_bars` or `conf_intervals`)
```

Comment:
Once the API is finalized, please review the doc string to ensure that it all matches up and is clear.